src/Akka.Persistence.Sql/Extensions/ConnectionFactoryExtensions.cs

```diff
@@ -72,5 +73,44 @@ public static class ConnectionFactoryExtensions
                 throw;
             }
         }
+        
+        internal static Task<T> ExecuteWithTransactionAsync<TState,T>(
+            this DbStateHolder factory,
+            TState state,
+            Func<AkkaDataConnection, CancellationToken, TState, Task<T>> handler)
+        {
+            return factory.ConnectionFactory.ExecuteWithTransactionAsync(state, factory.IsolationLevel, factory.ShutdownToken, handler);
+        }
```

Comment:
Convenience method; since we are using the stateholder to avoid other captures, may as well add some sugar to go with it and make the main DAO cleaner.